### ISSUE TITLE
[BUG]: predict_model output show unusual high number in error metrics

### ISSUE THREAD

**Issue by residual-07** [pycaret/pycaret#3417]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [ ] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

In previous version of pycaret, predict_model(), was giving me realistic error metrics number where i was conducting further testings. However, after the update, I'm getting unusual higher error rate in predict_model()

### Reproducible Example

```python
from pycaret.regression import *

# import the regression module
#add categorilac_features if there is string present in data
from pycaret.regression import *


# initialize setup
s = setup(data = df,
          train_size = 0.9,
          data_split_shuffle  = False, 
          target = 'MIPS',
          fold_strategy = 'timeseries',
          transformation = True, 
          remove_multicollinearity = True, 
          multicollinearity_threshold = 0.95,
          numeric_features = [ 'Hour', 'Week', 'Day', 'Month', 'Year', 'trend', 'cycle',
       'UPCharge', 'DownCharge'],
          
          fold = 3,
          transform_target = True,
          session_id = 123)


lgbm1 = create_model('xgboost')
Results: 

  	MAE 	MSE 	RMSE 	R2 	RMSLE 	MAPE
Fold 	  	  	  	  	  	 
0 	5.7885 	92.2749 	9.6060 	0.9976 	0.0150 	0.0078
1 	6.7526 	184.8536 	13.5961 	0.9960 	0.0223 	0.0086
2 	6.9182 	174.5903 	13.2133 	0.9963 	0.0188 	0.0081
Mean 	6.4864 	150.5730 	12.1384 	0.9966 	0.0187 	0.0082
Std 	0.4981 	41.4353 	1.7975 	0.0007 	0.0030 	0.0003
```

### Expected Behavior

`prediction_HOLDOUT = predict_model(lgbm1)`

                                            Model | MAE | MSE | RMSE | R2 | RMSLE | MAPE
-- | -- | -- | -- | -- | -- | ---- | -- | -- | -- | -- | -- | ---- | -- | -- | -- | -- | -- | ---- | -- | -- | -- | -- | -- | --
Light Gradient Boosting Machine | 4.0246 | 88.7856 | 9.4226 | 0.9982 | 0.0141 | 0.0052

### Actual Results

```python-traceback
And this is the output from new version: 

prediction_HOLDOUT = predict_model(lgbm1)


  Model 	MAE 	MSE 	  RMSE 	           R2 	        RMSLE 	  MAPE
0 Lgbm       823.4671 718132.2214 847.4268 	-70181.6877 	3.2365 	25.3977
```

### Installed Versions

<details>
PyCaret required dependencies:
                 pip: 21.2.4
          setuptools: 60.10.0
             pycaret: 3.0.0
             IPython: 7.34.0
          ipywidgets: 7.7.2
                tqdm: 4.64.0
               numpy: 1.21.6
              pandas: 1.4.2
              jinja2: 3.0.3
               scipy: 1.8.1
              joblib: 1.2.0
             sklearn: 1.1.1
                pyod: 1.0.9
            imblearn: 0.9.1
   category_encoders: 2.5.0
            lightgbm: 3.3.2
               numba: 0.55.2
            requests: 2.27.1
          matplotlib: 3.5.2
          scikitplot: 0.3.7
         yellowbrick: 1.4
              plotly: 5.13.1
             kaleido: 0.2.1
         statsmodels: 0.13.5
              sktime: 0.16.1
               tbats: 1.1.0
            pmdarima: 1.8.5
              psutil: 5.9.1

PyCaret optional dependencies:
                shap: 0.41.0
           interpret: 0.3.2
                umap: 0.5.3
    pandas_profiling: 3.6.6
  explainerdashboard: 0.4.1.1
             autoviz: 0.1.58
           fairlearn: 0.8.0
             xgboost: 1.7.4
            catboost: 1.1.1
              kmodes: 0.12.2
             mlxtend: 0.21.0
       statsforecast: 1.5.0
        tune_sklearn: 0.4.5
                 ray: 2.3.0
            hyperopt: 0.2.7
              optuna: 3.1.0
               skopt: 0.9.0
              mlflow: Not installed
              gradio: 3.8.1
             fastapi: 0.85.1
             uvicorn: 0.19.0
              m2cgen: 0.10.0
           evidently: Not installed
               fugue: 0.8.1
           streamlit: 1.20.0
             prophet: 1.1.2

</details>

**Comment by moezali1** [comment]
@residual-07  Please provide the reproducible notebook with data for us to troubleshoot.

**Comment by residual-07** [comment]
Notebook example 
```import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import os
os.chdir("/Users/name/Desktop")
import sys
from sklearn.model_selection import train_test_split
#os.environ["CUDA_DEVICE_ORDER"]="PCI_BUS_ID"
#os.environ["CUDA_VISIBLE_DEVICES"]="1"
import warnings
warnings.filterwarnings("ignore")
from datetime import datetime

df = pd.read_csv('https://github.com/pycaret/pycaret/files/11018013/data_git.csv')
df['Dates'] = pd.to_datetime(df['Dates'],infer_datetime_format=True)

#Feature Engineering 
# create 12 month moving average
df['MA_1Day'] = df['MIPS'].rolling(24).mean()

##############################################

#replace 0 with mean
df = df.replace({'MIPS': 0}, df['MIPS'].mean())

# create 12 month moving average
df['MA_1Day'] = df['MIPS'].rolling(24).mean()


#Date & time decomponsition

df['Hour'] = [i.hour for i in df['Dates']]
df['Week'] = [i.week for i in df['Dates']]
df['Day'] = [i.day for i in df['Dates']]
df['Month'] = [i.month for i in df['Dates']]
df['Year'] = [i.year for i in df['Dates']]



###############################################
# Tuple unpacking (hodrick prescot filter)
from statsmodels.tsa.filters.hp_filter import hpfilter

mips_trend, mips_cycle = hpfilter(df['MIPS'], lamb=1000600)

df['trend'] = mips_trend #The trend provides information on longer-term movements in the seasonally adjusted data series over several years.
df['cycle'] = mips_cycle #

##################################################
#Dummy variables for outliers

df['UPCharge'] = np.where( df['MIPS'] > 1000, 1, 0)
df['DownCharge'] = np.where( df['MIPS'] < 500, -1, 0)

#Remove columns to proceed to modeling of regression
# drop unnecessary columns and re-arrange
df.drop(['Dates', 'MA_1Day'], axis=1, inplace=True)
#df = df[['Year','Month','Day', 'Hour','UPCharge','DownCharge','cycle', 'MIPS']]

# import the regression module
#add categorilac_features if there is string present in data
from pycaret.regression import *


# initialize setup
s = setup(data = df,
          train_size = 0.9,
          data_split_shuffle  = False, 
          target = 'MIPS',
          fold_strategy = 'timeseries',
          transformation = True, 
          remove_multicollinearity = True, 
          multicollinearity_threshold = 0.95,
          numeric_features = [ 'Hour', 'Week', 'Day', 'Month', 'Year', 'trend', 'cycle',
       'UPCharge', 'DownCharge'],
          
          fold = 3,
          transform_target = True,
          session_id = 123)

#verify score of training & testing and check if additional tunning is necessary
#lightgbm = create_model('lightgbm')
lgbm1 = create_model('lightgbm')

prediction_HOLDOUT = predict_model(lgbm1)
[data_git.csv](https://github.com/pycaret/pycaret/files/11018013/data_git.csv)


```

**Comment by ngupta23** [comment]
Now that we have a dedicated time series module, it may also be worthwhile to have a look at that (it required less feature engineering on your end). I converted your code into that module and the results are reasonable. Please have a look (you can adjust the settings per your need).

Of course, we should continue to look at what is going on with the regression module and why you are getting those large errors metrics, but at least this can get you going till then.

```python
# !pip install pycaret

import numpy as np
import pandas as pd
from datetime import datetime
from pycaret.time_series import TSForecastingExperiment

# Read data
df = pd.read_csv('https://github.com/pycaret/pycaret/files/11018013/data_git.csv')
df['Dates'] = pd.to_datetime(df['Dates'],infer_datetime_format=True)
df.head()

# Set index
df = df.set_index("Dates")

# Convert index to period index (pycaret works best with PeriodIndex)
df.index = df.index.to_period(freq='H')

# Add missing indices
idx = pd.period_range(min(df.index), max(df.index))
df = df.reindex(idx, fill_value=np.nan)

df.info()

exp = TSForecastingExperiment()
exp.setup(data=df, target = "MIPS", fold=3, fh=72, numeric_imputation_target="ffill", session_id=42)

model = exp.create_model("lightgbm_cds_dt")

exp.plot_model(model)
```

![image](https://user-images.githubusercontent.com/33585645/226380317-a1a02d40-3301-41d1-8733-9a49e3a98a3d.png)

**Comment by residual-07** [comment]
Thank you @ngupta23 for providing me alternative solution. I will take look into that!

**Comment by moezali1** [comment]
Closing as part of the 4.0 backlog cleanup — see #4177.

PyCaret 4.0 is a ground-up rewrite (OOP-only API, sklearn-native engine, Python 3.11+, sklearn ≥ 1.7) and almost all pre-4.0 issues either reference removed APIs (`setup()`, `plot_model`, `evaluate_model`, `interpret_model`, `automl`, the legacy time-series module, the Streamlit dashboard, …) or no longer reproduce on the new code path.

**If this still affects you in 4.0, please re-file as a new issue** with:

- the experiment constructor call (`Experiment(...)`),
- the operation that fails,
- the full traceback,
- your Python / sklearn / pycaret versions.

Resources for 4.0:

- Docs: https://pycaret.org/docs
- Migrate from 3.x: https://pycaret.org/docs/guides/migrating-from-3-x
- API reference: https://pycaret.org/reference
- FAQ: https://pycaret.org/docs/resources/faqs

Thanks for filing — and apologies for the long latency. The repo had drifted; we're starting clean.
